application/src/main/scala/com/azavea/franklin/extensions/validation/package.scala

```diff
@@ -0,0 +1,31 @@
+package com.azavea.franklin.extensions
+
+import cats.implicits._
+import com.azavea.stac4s.extensions.label.{LabelItemExtension, LabelLinkExtension}
+import com.azavea.stac4s.extensions.layer.LayerItemExtension
+import com.azavea.stac4s.{StacItem, StacLink}
+import monocle.macros.GenLens
+
+package object validation {
+
+  def getItemValidator(extensions: List[ExtensionName]): StacItem => StacItem = {
+    makeValidator(extensions map {
+      case Label        => ItemValidator[LabelItemExtension].validate _
+      case Layer        => ItemValidator[LayerItemExtension].validate _
+      case Unchecked(_) => (x: StacItem) => x
+    })
+  }
+
+  def getLinkValidator(extensions: List[ExtensionName]): StacLink => StacLink = {
+    makeValidator(extensions map {
+      case Label        => LinkValidator[LabelLinkExtension].validate _
+      case Layer        => (x: StacLink) => x
+      case Unchecked(_) => (x: StacLink) => x
+    })
+  }
+
+  private def makeValidator[A](funcs: List[A => A]): A => A =
+    funcs.toNel map { _.reduce((f1: A => A, f2: A => A) => f1 compose f2) } getOrElse { x => x }
+
+  val linksLens = GenLens[StacItem](_.links)
```

Comment:
I threw it in here because I knew it was going to be `.copy(links=foo.links.copy(...))` -- I'm not sure about including it in Stac4s. I know doobie ships some optics on transactors, which I think makes sense, but I'm not sure what it should look like in library code in general. Maybe an `optics` package like circe has?

Reply:
Yeah, I like the idea of an `optics` package in `stac4s` -- it could maybe start small. It [looks](https://github.com/azavea/franklin/search?l=Scala&q=%60.copy%60) like the most common operation with `.copy` in Franklin right now is for links. I think as we start doing more data intake/manipulation we're going to find it useful to more easily modify nested attributes. I don't want to prematurely optimize though at the same time.

Reply:
azavea/stac4s#113